geronimo-openapi-impl/src/main/java/org/apache/geronimo/microprofile/openapi/impl/processor/AnnotationProcessor.java

```diff
@@ -307,10 +308,17 @@ private Operation buildOperation(final OpenAPI api, final AnnotatedMethodElement
                      .filter(v -> !v.isEmpty())
                      .collect(toList()))
                 .ifPresent(operation::tags);
-
-        of(m.getAnnotationsByType(APIResponse.class)).filter(s -> s.length > 0).ifPresent(items -> {
+        
+        final Collection<APIResponse> apiResponses = Stream
```

Comment:
hmm, here the getAnnotationsByType usage was supposed to catch it since it is repeatable, no? - https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Executable.html#getAnnotationsByType-java.lang.Class-.

Also please add a test even if trivial, seems it is not caught by tck if previous point is not the issue you hit - bad API?

Reply:
You are very right, the `@APIResponse` is repetable and `@APIResponses` are properly handled, turned out to be an issue on my side. Thank you!